torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2263,8 +2263,31 @@ def sample_inputs_fliplr_flipud(op_info, device, dtype, requires_grad, **kwargs)
 # TODO: clamp shares tensors among its sample inputs --- we should prohibit this!
 def sample_inputs_clamp(op_info, device, dtype, requires_grad, **kwargs):
     tensors = (
-        make_tensor((2, 3, 2), device=device, dtype=dtype, low=None, high=None, requires_grad=requires_grad),
-        make_tensor((2, 0, 3), device=device, dtype=dtype, low=None, high=None, requires_grad=requires_grad),
+        make_tensor((S, M, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
+        make_tensor((S, M, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
+        make_tensor((S, M, S), device, dtype, low=None, high=None, requires_grad=requires_grad),
+    )
+
+    requires_grad = tensors[0].requires_grad
+    lb = torch.empty((S, M, S), device=device, dtype=dtype, requires_grad=requires_grad)
+    ub = torch.empty((S, M, S), device=device, dtype=dtype, requires_grad=requires_grad)
+    with torch.no_grad():
+        lb[:] = tensors[1].min(tensors[2])
+        ub[:] = tensors[1].max(tensors[2])
+
+    def detach(tensor):
+        return tensor.clone().detach_().requires_grad_(requires_grad)
+
+    return [
+        SampleInput(tensors[0], args=(lb, ub)),
```

Comment:
I think the input tensor for each sample input also needs to be cloned in a no grad context if you take this approach, otherwise you'll have a relationship between tensors in different sample inputs